storage/src/main/java/org/apache/uniffle/storage/handler/impl/ComposedClientReadHandler.java

```diff
@@ -144,44 +115,9 @@ public ShuffleDataResult readShuffleData() {
     return shuffleDataResult;
   }
 
-  private String getCurrentHandlerName() {
-    String name = "UNKNOWN";
-    switch (currentHandler) {
-      case HOT:
-        name = "HOT";
-        break;
-      case WARM:
-        name = "WARM";
-        break;
-      case COLD:
-        name = "COLD";
-        break;
-      case FROZEN:
-        name = "FROZEN";
-        break;
-      default:
-        break;
-    }
-    return name;
-  }
-
   @Override
   public void close() {
-    if (hotDataReadHandler != null) {
-      hotDataReadHandler.close();
-    }
-
-    if (warmDataReadHandler != null) {
-      warmDataReadHandler.close();
-    }
-
-    if (coldDataReadHandler != null) {
-      coldDataReadHandler.close();
-    }
-
-    if (frozenDataReadHandler != null) {
-      frozenDataReadHandler.close();
-    }
+    handlerMap.values().stream().filter(Objects::nonNull).forEach(ClientReadHandler::close);
```

Comment:
Is `filter(Objects::nonNull)` necessary?

Reply:
Not required, but just in case.
And to be consistent with the old logic.

Reply:
I think we can remove it here, because there is an extra loop. WDYT?

Reply:
There are at most 4 elements in the map ...

Reply:
The key is whether it is necessary to keep it. It's OK for me too to keep it.

Reply:
It's just for safety.

Reply:
Actually, the values of `EnumMap` will not be null. If we put `null` to `EnumMap`, it will be modified to an instance of `Object`.

Reply:
OK, let me check.

Reply:
> Actually, the values of `EnumMap` will not be null. If we put `null` to `EnumMap`, it will be modified to an instance of `Object`.

No, it could be `null`. The following test passed.

<img width="663" alt="test" src="https://user-images.githubusercontent.com/5821159/217992019-510a2a52-8688-4d8e-87d5-3cac1b6f0845.png">

Reply:
My mistake. When we get value from `EnumMap`, it will be modified to null again.